client/state/routing/middleware.js

```diff
@@ -9,11 +9,18 @@ import page from 'page';
  */
 import localforage from 'lib/localforage';
 import { isOutsideCalypso } from 'lib/url';
-import { ROUTE_SET } from 'state/action-types';
+import { ROUTE_SET, ROUTE_TO } from 'state/action-types';
 
 const debug = debugFactory( 'calypso:restore-last-location' );
 const LAST_PATH = 'last_path';
 
+export const routingMiddleware = () => ( next ) => ( action ) => {
+	if ( action.type === ROUTE_TO ) {
+		page( action.path );
+	}
```

Comment:
it's at this point I wonder if we could trap on `COMMENTS_SELECT_FILTER` to follow-up with the appropriate routing